File: backend/nodejs/apps/src/libs/services/telemetry/identity.ts
/**
 * Identity helpers for product-behaviour events.
 *
 * Events are tagged with the caller's email-domain so adoption / funnel /
 * engagement dashboards can exclude internal (PipesHub) traffic. Email/domain
 * are identity, not document/query content, so they are allowed by the
 * collector.
 */

/** Lower-cased domain part of an email, or `"unknown"` when absent. */
export function domainFromEmail(email: string | undefined | null): string {
  if (!email || !email.includes('@')) return 'unknown';
  return email.split('@').pop()?.trim().toLowerCase() || 'unknown';
}

export function normalizeOrgId(orgId: unknown): string {
  if (orgId == null) return 'unknown';
  const str = String(orgId);
  return str !== '' && str !== '[object Object]' ? str : 'unknown';
}

File: backend/python/app/telemetry/identity.py
from typing import Optional


def domain_from_email(email: Optional[str]) -> str:
    """Lower-cased domain part of an email, or ``"unknown"`` when absent."""
    if not email or "@" not in email:
        return "unknown"
    return email.rsplit("@", 1)[1].strip().lower() or "unknown"
